api/src/main/java/org/openmrs/module/registrationcore/api/mpi/pixpdq/Hl7SenderHolder.java

```diff
@@ -0,0 +1,31 @@
+package org.openmrs.module.registrationcore.api.mpi.pixpdq;
+
+
+import org.openmrs.api.APIException;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.registrationcore.RegistrationCoreConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+
+public class Hl7SenderHolder {
```

Comment:
Since this class is named holder, I'd expect it to cache a copy of the bean that way you don't have to look it up from the application context every time.

Reply:
Ok, I have changed the class in the following way, does it look correct to you/ is this what you meant?

public class Hl7SenderHolder {

    @Autowired
    private ApplicationContext context;

    Hl7v2Sender hl7v2Sender;
    
    public Hl7v2Sender getHl7v2Sender() {
        if (hl7v2Sender == null){
            initHl7v2Sender();
        }
        return hl7v2Sender;
    }

    public Hl7v2Sender initHl7v2Sender() {
        String propertyName = RegistrationCoreConstants.GP_MPI_HL7_IMPLEMENTATION;
        Object bean;
        try {
            String beanId = Context.getAdministrationService().getGlobalProperty(propertyName);
            bean = context.getBean(beanId);
        } catch (APIException e) {
            throw new APIException(e);
        }
        if (!(bean instanceof Hl7v2Sender))
            throw new IllegalArgumentException(propertyName
                    + " must point to bean implementing Hl7v2Sender");

        return (Hl7v2Sender) bean;
    }
}

Reply:
I guess I was implying renaming it instead so that it doesn't signal that it's a Holder of the bean, sorry about that if I wasn't clear

Reply:
ok so what should it be renamed to?

Reply:
To be honest, I don't see the reason why this class exists in the first place, all it does is look up the spring bean matching the GP value, you have several of such beans I think, in theory I'd expect a single utility class that has a method that returns a spring bean given a GP name, this promotes reusability.

Reply:
Did you see this comment? Why dow e have this class anyways?